lib/elsa/producer.ex

```diff
@@ -1,34 +1,25 @@
 defmodule Elsa.Producer do
   @moduledoc """
-  Defines functions to manage producer supervisors and works,
-  as well as send messages to topics.
+  Defines functions to send messages to topics based on either a list of endpoints or a named client.
   """
-
-  def start_producer(endpoints, topic, config \\ []) do
-    name = Keyword.get(config, :name, Elsa.default_client())
-
-    start_client(endpoints, name)
-    :brod.start_producer(name, topic, config)
-  end
-
-  def stop_producer(client, topic), do: :brod_client.stop_producer(client, topic)
-
   def produce_sync(client \\ Elsa.default_client(), topic, partition \\ 0, key \\ "ignored", value)
 
   def produce_sync(endpoints, topic, partition, key, value) when is_list(endpoints) do
     client = Elsa.default_client()
 
-    start_producer(endpoints, topic, name: client)
+    Elsa.Producer.Supervisor.start_producer(endpoints, topic, name: client)
     produce_sync(client, topic, partition, key, value)
   end
 
-  def produce_sync(client, topic, partition, key, value) do
-    :brod.produce_sync(client, topic, partition, key, value)
-  end
+  def produce_sync(client, topic, partitioner, key, value) do
+    partition =
```

Comment:
What do you think about using a guard on `partitioner` here? Something like:

```elixir
def produce_sync(client, topic, partitioner, key, value) when is_atom(partitioner) do
  {:ok, partition_num} = :brod.get_partitions_count(client, topic)
  partition = Elsa.Producer.Partitioner.partition(partitioner, partition_num, value)
  produce_sync(client, topic, partition, key, value)
end

def produce_sync(client, topic, partition, key, value) do
  :brod.produce_sync(client, topic, partition, key, value)
end
```

I like this better for a few reasons:
1. I prefer guards to conditional blocks
2. I like pattern matching against `Partitioner.partition/3` over applying an atom to `Partitioner` with two arguments.
3. Recursion!

It would require the next few changes in `Partitioner`...